components/homme/src/test_src/dry_planar.F90

```diff
@@ -170,11 +172,141 @@ subroutine planar_rising_bubble_init(elem,hybrid,hvcoord,nets,nete)
 
 end subroutine planar_rising_bubble_init
 
 
+subroutine dry_bubble_init(elem,hybrid,hvcoord,nets,nete,d,f)
+
+  use control_mod, only: bubble_T0, bubble_dT, bubble_xycenter, bubble_zcenter, bubble_ztop, &
+                         bubble_radius, bubble_cosine, &
+                         Lx, Ly, Sx, Sy
+
+  type(element_t),    intent(inout), target :: elem(:)                  ! element array
+  type(hybrid_t),     intent(in)            :: hybrid                   ! hybrid parallel structure
+  type(hvcoord_t),    intent(inout)         :: hvcoord                  ! hybrid vertical coordinates
+  integer,            intent(in)            :: nets,nete                ! start, end element index
+  real(rl),           intent(in)            :: d                        ! radius of perturbation
+  real(rl),           intent(in)            :: f                        ! (const) Coriolis force
+
+  integer :: i,j,k,ie                                                   ! loopindices
+  real(rl):: x,y,hyam,hybm,hyai,hybi           
+  real(rl):: pi(nlevp), dpm(nlev), th0(nlevp), th0m(nlev), ai(nlevp), bi(nlevp), rr
+
+#ifdef MODEL_THETA_L
+
+  if (hybrid%masterthread) then
+     write(iulog,*) 'initializing hot bubble with'
+     print *, 'Lx, Ly =', Lx, Ly
+     print *, 'Sx, Sy =', Sx, Sy
+     print *, 'bubble_T0',  bubble_T0
+     print *, 'bubble_dT', bubble_dT
+     print *, 'bubble_xycenter', bubble_xycenter
+     print *, 'bubble_zcenter', bubble_zcenter
+     print *, 'bubble_ztop', bubble_ztop
+     print *, 'bubble_radius', bubble_radius
+     print *, 'bubble_cosine', bubble_cosine
+  endif
+
+  !evenly spaced with reversed indexing, just like in homme eta coord
+  call get_evenly_spaced_z(zi,zm, 0.0_rl,bubble_ztop)
+
+  ! set initial conditions
+  do ie = nets,nete
+    do j=1,np; do i=1,np
+
+      ! get horizontal coordinates at column i,j
+      x  = elem(ie)%spherep(i,j)%lon
+      y  = elem(ie)%spherep(i,j)%lat
+
+      do k=1,nlevp
+
+        rr = sqrt( (x-bubble_xycenter)   *(x-bubble_xycenter)+&
+                   (y-bubble_xycenter)   *(y-bubble_xycenter)+&
+                   (zi(k)-bubble_zcenter)*(zi(k)-bubble_zcenter) )
+
+        !set pot. temperature on interfaces
+        if ( rr < bubble_radius ) then 
+          if (bubble_cosine) then
+            th0(k) = bubble_T0 + bubble_dT * cos(rr*dd_pi / bubble_radius / 2.d0)
+          else
+            th0(k) = bubble_T0 + bubble_dT
+          endif
+
+        else
+          th0(k) = bubble_T0
+        endif
+
+        !set interface pressure from const lapse rate, does not depend on x,y, or _dT
+        pi(k) =  p0*( (270.0 - zi(k)*g/cp)/270.0  )**(1.0/kappa)
+      enddo
+
+      elem(ie)%state%ps_v(i,j,:) = pi(nlevp)
+      elem(ie)%state%v = 0.0
+      elem(ie)%state%w_i= 0.0
+
+      do k=1,nlev
+        !set pottemp, dp, other state vars on midlevels
+        th0m(k)=(th0(k)+th0(k+1))/2.0
+        dpm(k)=(pi(k+1)-pi(k))
```

Comment:
Why not directly set theta and pseudo-density, since there are formulas that exist for them? My concern is this linear interpolation to going to lead to an discrete imbalance between the background theta0 and pseudo-density fields, when really they should be in discrete hydrostatic balance.

Reply:
For the hydrostatic balance, the code was updated to compute new zi from given pressure/dp and const theta T0 for the reference profile though I do not see any differences in the dry bubble solutions. The code you point to is a perturbation which should not be in the hydro balance. Moreover, since it is a perturbation, we can average and it will only affect the shape, nothing else. To do it right, I think I would have to follow discrete averaging rules for homme. For example, for phi on midlevels it is a simple average, but for theta on interfaces it is a weighted average as in Taylor2020 eqn 48. I would have to invert eqn 48, which seems not needed here.